### PR TITLE
Assert fix

### PR DESCRIPTION
**Pull request by AngelOnFira**

### CODE REVIEW

**Comment by AngelOnFira** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #33** <a href="https://app.graphite.dev/github/pr/rivet-gg/plugin-godot/33?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #34** <a href="https://app.graphite.dev/github/pr/rivet-gg/plugin-godot/34?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rivet-gg/plugin-godot/33?utm_source=stack-comment).

File: addons/rivet/rivet_helper.gd [hunk 1]
```diff
@@ -115,3 +115,7 @@ func _rivet_player_connect_failed(error, id: int):
 
 func rivet_print(message: String):
 	print("[Rivet] %s" % message)
+
+func _assert(condition: bool, message: String = "Assertion failed"):
+	if not condition:
+		OS.crash(message)
```
[the remark this answers is not in the document]
Comment (AngelOnFira) [reply addons/rivet/rivet_helper.gd:121]: Good point, I didn't think about it like that. However, there are some cases where something bad happens that would be difficult to recover from, or could lead to other hidden bugs if the server doesn't crash. It's not entirely clear where these might come up the most (though I think weird things might happen if you, say, call certain Rivet API endpoints multiple times from the same server). For these, there should probably be some kind of check that does crash if the condition isn't true.

For now, I'll change it to just print the warning, but we can look at adding another method if needed.

[the remark this answers is not in the document]
Comment (NathanFlurry) [reply addons/rivet/rivet_helper.gd]: gdscript macros when

in all seriousness, i think we had talked about creating the equivalent of the hub's error toast in the godot plugin at one point. it sounds like we need to do some go-style error handling or create our own Result type.

can we create a ticket about this @AngelOnFira?

**Review by NathanFlurry — APPROVED** [review]

File: addons/rivet/rivet_helper.gd (hunk 1 above)
[the remark this answers is not in the document]
Comment (AngelOnFira) [reply addons/rivet/rivet_helper.gd]: I think we have some kind of error toast, but the asserts are mainly being called on server stuff I think. I'm haven't thought too much about client error handling yet, will need to go over that. Would love a better result option :P

[the remark this answers is not in the document]
Comment (AngelOnFira) [reply addons/rivet/rivet_helper.gd]: https://linear.app/rivet-gg/issue/GDT-96/hub-button-should-go-to-linked-hub-in-bootstrap

**Pull request merged**